src/API/Google/Settings.php

```diff
@@ -342,10 +355,7 @@ protected function create_shipping_service( string $country, string $method, str
 		$rate_groups[] = $this->create_rate_group_object( $currency, $rate, $method );
 		$service->setRateGroups( $rate_groups );
 
-		$times = $this->get_times();
-		if ( array_key_exists( $country, $times ) ) {
-			$service->setDeliveryTime( $this->create_time_object( intval( $times[ $country ] ) ) );
-		}
```

Comment:
❓ 

Could we not throw the error here? 

```
		if ( array_key_exists( $country, $times ) ) {
			$service->setDeliveryTime( $this->create_time_object( intval( $times[ $country ] ) ) );
		}
       else{
         throw new InvalidValue( 'No estimated shipping time found for country: ' . $country );
       }
```

And the rest of the logic could stay as it was.

Reply:
We could but then there might be cases where we don't sync anything to Google and still return a successful response. In theory, if this exception is thrown it means there's a logic error in our code because in https://github.com/woocommerce/google-listings-and-ads/issues/1380 we will sync the rates only if a shipping time is set. Let me know what you think.

Reply:
> We could but then there might be cases where we don't sync anything to Google and still return a successful response

Could you elaborate on this a bit more? As far as I understand, having the current implementation of this PR or throwing the exception in `create_shipping_service` would have the same behaviour. If one of the rates does not have a time, it will throw an exception and it is not going to sync any rate.

Reply:
Hmm. I'm not sure I understand. Here we only set the delivery time. The actual exception is thrown in the `create_shipping_service`. Do you mean we should move that validation to here?

Reply:
Yes, when I was doing the review I was wondering what was the reason to do the validation in `update_shipping_rates` and not in `create_shipping_service`. If we did it in `create_shipping_service` would not be necessary to add the argument `delivery_days` to the methods: `create_shipping_service` and `create_conditional_free_shipping_service` and all the logic could stay the same as it was before.

Reply:
My main reason for refactoring those methods was that the `get_times` method was being called for every country and now it can be called only once. It was unnecessary to query the DB on every for loop where it could've been done just once.

Reply:
Aaaah! Thanks for the explanation, now it makes sense. My point of view was more from "throwing the exception if there is no time set" but I did not realize that the issue with `get_times` was there before this PR.